plugins/destination/mysql/client/schema.go

```diff
@@ -63,6 +71,10 @@ func (c *Client) getTableColumns(ctx context.Context, tableName string) ([]schem
 		if constraintType != nil {
 			primaryKey = strings.Contains(*constraintType, "PRIMARY KEY")
 		}
+		// subpart only non nil for pks on blob/text columns
+		if subpart != nil && *subpart != maxPrefixLength {
+			primaryKey = false
```

Comment:
1. pre migration:
* table `A` PK includes column named `col`, type `text(64)`

2. new schema doesn't include `col` as part of the PK

3. we verify that the `(191)` != `(64)`, so `primaryKey = false` will be executed

Does it mean that the migration in safe mode will succeed?
Is it OK?

Reply:
Technically this could be a `safe` migration as increasing the prefix length used in a PK is fine. But no in `migrate_mode: safe` this should fail.

The column isn't of type `text(64)` it is of type `text`... The only part that includes the prefix length is the field `subpart`...